### PR TITLE
Exit when unable get self-sign cert

### PR DESCRIPTION
**Pull request by jhadvig**
@benjaminapetersen PTAL

### CODE REVIEW

**Comment by benjaminapetersen** [comment]
@jhadvig So should we force a fail, or do we want to support the possibility of allowing the OAuth flow?  Maybe with an environment variable `--OAUTH` we could skip this check?

**Comment by jhadvig** [comment]
@benjaminapetersen definitelly. Opened this PR to discuss the all the possible flow that we want cover, from obtaining self-sign certs, to oauth...
Lets chat to get a clear picture to solve this :)

**Comment by jhadvig** [comment]
@benjaminapetersen let's talk today regarding this, so we align on the different flows when we will need the self-sign certificate and we won't need it.

**Comment by jhadvig** [comment]
@benjaminapetersen I've updated the PR. Refactored it a bit so now obtaining the cert will be optional (by default will be obtained), since when in use in production cluster we probably wont need to get the cert and store it.
PTAL

File: smoke-test.sh [hunk 1]
```diff
@@ -15,11 +15,18 @@ host_with_port=$(echo "$url" | sed 's/\/.*$//')
 host=$(echo "$host_with_port" | sed 's/:.*$//')
 port=$(echo "$host_with_port" | grep : | sed -e 's,^.*:,:,g' -e 's,.*:\([0-9]*\).*,\1,g')
 
-os::log::info "Getting certificate for ${CONSOLE_URL}"
-timeout 30 google-chrome --no-sandbox --headless --disable-gpu --dump-dom "$CONSOLE_URL" &>/dev/null
-echo 'GET /' | timeout 30 openssl s_client -showcerts -connect "$host:$port" | openssl x509 -outform PEM > /tmp/console-e2e.pem
-os::log::info "Storing obtained certificate"
-certutil -d "sql:$HOME/.pki/nssdb" -A -n console -t Pu,, -i /tmp/console-e2e.pem
+if [ "$SELF_SIGN_CERT" = true ]; then
+  os::log::info "Getting certificate for ${CONSOLE_URL}"
```
Comment (benjaminapetersen) [inline smoke-test.sh]: Shall we update the log to "Getting self signed certificate...." just to help ensure this is super clear?

**Comment by benjaminapetersen** [comment]
@jhadvig nice

**Pull request merged**